rl/agents/sarsa.py

```diff
@@ -0,0 +1,471 @@
+from rl.agents.dqn import DQNAgent
+import warnings
+from copy import deepcopy
+
+import numpy as np
+from keras.callbacks import History
+
+from rl.callbacks import TestLogger, TrainEpisodeLogger, TrainIntervalLogger, Visualizer, CallbackList
+from rl.agents.dqn import mean_q
+from rl.util import huber_loss
+from keras.layers import Input, Lambda
+from keras.models import Model
+import keras.backend as K
+
+
+class Sarsa(DQNAgent):
```

Comment:
SARSA shouldn't inherit from `DQNAgent` since we pretty much don't use anything that it offers to begin with. I think it would be a lot cleaner to inherit from `Agent` directly and maybe re-implement the few methods that are shared between DQN and SARSA. We could also create a common super-class if the shared code is enough.